mangaki/mangaki/api/user.py

```diff
@@ -90,5 +97,42 @@ def export_user_data(request: Request):
             builder.cleanup()
             return sendfile(request, archive.local_archive.path)
         except (OSError, IOError, django.db.Error) as e:
-            print(e)
             return Response({}, status=status.HTTP_503_SERVICE_UNAVAILABLE)
+
+
+@api_view(['GET'])
+@permission_classes((AllowAny,))
+def get_user_and_friends_positions(request: Request, algo_name):
+    """
+    Compute the position of user and friends on the map
+    """
+    rated_works = current_user_ratings(request)
+    df_mine = pd.DataFrame(rated_works.items(), columns=('work_id', 'choice'))
+    df_mine['user_id'] = request.user.id if not request.user.is_anonymous else -1
+    df_item_pos = pd.DataFrame(
+        get_2d_embeddings(algo_name)['works']).set_index('work_id')
+    available_works = df_item_pos.index
+
+    if not request.user.is_anonymous:
+        friend_ratings = friend_ratings(request)
```

Comment:
Isn't this missing friend_ids argument?

Reply:
Indeed and also it was overriding the function with the variable :roll_eyes: